### PR TITLE
Aryanmajeed/issue51

### LINKED ISSUE #51: Connect the search page.
**Is your feature request related to a problem? Please describe.**
N/A

**Describe how the solution should behave**
Fetch the data on the search page, and store it in the redux to a local state to apply filters too. Once the user chooses filters, apply them to the local state to see the correct data.

**Screenshots of the design/if any**
image

**Additional context**
Use the project slice in redux, and have a thunk fun to fetch those data.

### PR DESCRIPTION
**Pull request by aryankarim**
## What?
Added the advanced search and sorting functionality.
## Why?
The search page should be able to filter and sort data. 
## How to use?
go to route `search/`
## Screenshots
screencapture-localhost-3000-search-2021-10-28-21_20_17

### CODE REVIEW

**Comment by aryankarim** [comment]
PS: The location filters do not work for now. I have to adjust the firebase's user collection to add that functionality.

**Review by EsraaY21 — APPROVED** [review]
Nice work!

File: pages/search/index.js [hunk 1]
```diff
@@ -16,78 +16,126 @@ import {
...
+    dispatch(getAllPosts());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  useEffect(() => {
+    dispatch(getAllPosts());
+    if (status === "success") {
+      setPosts(data);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [status]);
```
Comment (MuhannedNoman) [inline pages/search/index.js]: Why not in one useEffect?

**Review by MuhannedNoman — APPROVED** [review]
LGTM, just left a small comment.

**Review by HaiderAltufaily — APPROVED** [review]
LGTM! great work!

File: pages/search/index.js (hunk 1 above)
Comment (aryankarim) [reply pages/search/index.js]: Yup! that is extra!

**Pull request merged**